### ISSUE TITLE
Segmentation fault (core dumped) with string as input in sabyenc3.encode('hello') 

### ISSUE THREAD

**Issue by sanderjo** [sabnzbd/sabyenc#19]
Core dump with string as input for sabyenc3.encode():
```
$ python3 -c "import sabyenc3; sabyenc3.encode('hello') "
Segmentation fault (core dumped)
```

```
$ python3 -c "import sabyenc3; print(sabyenc3.__version__); sabyenc3.encode('hello') "
4.0.2
Segmentation fault (core dumped)
```

With bytes so b'hello' it goes well:

```
$ python3 -c "import sabyenc3; print(sabyenc3.__version__); sabyenc3.encode(b'hello') "
4.0.2
```

```
$ python3 -c "import sabyenc3; print(sabyenc3.__version__); print(sabyenc3.encode(b'hello')) "
4.0.2
(b'\x92\x8f\x96\x96\x99', 3387906425)
```

```
>>> type('hello')
<class 'str'>
>>> type(b'hello')
<class 'bytes'>
```

**Comment by sanderjo** [comment]
FWIW core dump stuff:

```
sander@witte2004:~$ ulimit -c unlimited
sander@witte2004:~$ ll core
ls: cannot access 'core': No such file or directory
sander@witte2004:~$ python3 -c "import sabyenc3; print(sabyenc3.__version__); print(sabyenc3.encode('hello')) "
4.0.2
Segmentation fault (core dumped)
sander@witte2004:~$ ll core
-rw------- 1 sander sander 72060928 apr 20 05:52 core
```


```
sander@witte2004:~$ file core
core: ELF 64-bit LSB core file, x86-64, version 1 (SYSV), SVR4-style, from 'python3 -c import sabyenc3; print(sabyenc3.__version__); print(sabyenc3.encode(', real uid: 1000, effective uid: 1000, real gid: 1000, effective gid: 1000, execfn: '/usr/bin/python3', platform: 'x86_64'
```


```
sander@witte2004:~$ gdb python3 core
GNU gdb (Ubuntu 9.2-0ubuntu1~20.04) 9.2
Copyright (C) 2020 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from python3...
(No debugging symbols found in python3)

warning: core file may not match specified executable file.
[New LWP 97557]
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
Core was generated by `python3 -c import sabyenc3; print(sabyenc3.__version__); print(sabyenc3.encode('.
Program terminated with signal SIGSEGV, Segmentation fault.
#0  encode_buffer (crc=<synthetic pointer>, bytes=<optimized out>, output_buffer=0x7f5206ae3010 "", input_buffer=0x0) at src/sabyenc3.c:657
657	src/sabyenc3.c: No such file or directory.
(gdb) where
#0  encode_buffer (crc=<synthetic pointer>, bytes=<optimized out>, output_buffer=0x7f5206ae3010 "", input_buffer=0x0) at src/sabyenc3.c:657
#1  encode (self=<optimized out>, args=<optimized out>) at src/sabyenc3.c:726
#2  0x00000000005f305a in PyCFunction_Call ()
#3  0x00000000005f3446 in _PyObject_MakeTpCall ()
#4  0x000000000056f600 in _PyEval_EvalFrameDefault ()
#5  0x000000000056822a in _PyEval_EvalCodeWithName ()
#6  0x000000000068c1e7 in PyEval_EvalCode ()
#7  0x000000000067d5a1 in ?? ()
#8  0x000000000067d61f in ?? ()
#9  0x000000000067d78f in PyRun_StringFlags ()
#10 0x000000000067d7ef in PyRun_SimpleStringFlags ()
#11 0x00000000006b61ec in Py_RunMain ()
#12 0x00000000006b64bd in Py_BytesMain ()
#13 0x00007f520b7790b3 in __libc_start_main (main=0x4eec80 <main>, argc=3, argv=0x7ffd08ba4368, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffd08ba4358)
    at ../csu/libc-start.c:308
#14 0x00000000005f927e in _start ()
(gdb) bt full
#0  encode_buffer (crc=<synthetic pointer>, bytes=<optimized out>, output_buffer=0x7f5206ae3010 "", input_buffer=0x0) at src/sabyenc3.c:657
        byte = <optimized out>
        encoded = <optimized out>
        in_ind = 0
        out_ind = 0
        col = 0
        encoded = <optimized out>
        in_ind = <optimized out>
        out_ind = <optimized out>
        col = <optimized out>
        byte = <optimized out>
#1  encode (self=<optimized out>, args=<optimized out>) at src/sabyenc3.c:726
        _save = 0x1da5610
        Py_input_string = 0x7f520ad47ab0
        Py_output_string = <optimized out>
        retval = 0x0
        input_buffer = 0x0
        output_buffer = 0x7f5206ae3010 ""
        input_len = <optimized out>
        output_len = 0
        crc = {crc = 4294967295, bytes = <optimized out>}
#2  0x00000000005f305a in PyCFunction_Call ()
No symbol table info available.
#3  0x00000000005f3446 in _PyObject_MakeTpCall ()
No symbol table info available.
#4  0x000000000056f600 in _PyEval_EvalFrameDefault ()
No symbol table info available.
#5  0x000000000056822a in _PyEval_EvalCodeWithName ()
No symbol table info available.
#6  0x000000000068c1e7 in PyEval_EvalCode ()
No symbol table info available.
#7  0x000000000067d5a1 in ?? ()
No symbol table info available.
#8  0x000000000067d61f in ?? ()
No symbol table info available.
#9  0x000000000067d78f in PyRun_StringFlags ()
No symbol table info available.
#10 0x000000000067d7ef in PyRun_SimpleStringFlags ()
No symbol table info available.
#11 0x00000000006b61ec in Py_RunMain ()
No symbol table info available.
#12 0x00000000006b64bd in Py_BytesMain ()
No symbol table info available.
#13 0x00007f520b7790b3 in __libc_start_main (main=0x4eec80 <main>, argc=3, argv=0x7ffd08ba4368, init=<optimized out>, fini=<optimized out>, rtld_fini=<optimized out>, stack_end=0x7ffd08ba4358)
    at ../csu/libc-start.c:308
        self = <optimized out>
        result = <optimized out>
        unwind_buf = {cancel_jmp_buf = {{jmp_buf = {7038160, 7956376582668929517, 6263376, 140724749878112, 0, 0, -7957672334482486803, -8012276781745059347}, mask_was_saved = 0}}, priv = {pad = {0x0, 
              0x0, 0x3, 0x7ffd08ba4368}, data = {prev = 0x0, cleanup = 0x0, canceltype = 3}}}
        not_first_call = <optimized out>
#14 0x00000000005f927e in _start ()
No symbol table info available.
(gdb) 
```

**Comment by Safihre** [comment]
It's a known issue, not one that I really care to fix 🚧

**Comment by Safihre** [comment]
Seems fixed by 5.0.
